rest-server/src/controllers/job.js

```diff
@@ -58,38 +47,27 @@ const init = (req, res, next) => {
   const jobName = req.body.jobName;
   new Job(jobName, (job, error) => {
     if (error) {
-      if (error.message === 'JobNotFound') {
+      if (error.code === 'ERR_NO_JOB') {
         req.job = job;
         next();
       } else {
-        logger.warn('internal server error');
-        return res.status(500).json({
-          error: 'InternalServerError',
-          message: 'internal server error',
-        });
+        return next(createError.unknown(error));
       }
     } else {
-      logger.warn('duplicate job %s', jobName);
-      return res.status(400).json({
-        error: 'DuplicateJobSubmission',
-        message: `job already exists: '${jobName}'`,
-      });
+      return next(createError('Conflict', 'ERR_CONFLICT_JOB', `Job name ${jobName} already exists`));
     }
   });
 };
 
 /**
  * Get list of jobs.
  */
-const list = (req, res) => {
+const list = (req, res, next) => {
   Job.prototype.getJobList(req._query, (jobList, err) => {
     if (err) {
-      logger.warn('list jobs error\n%s', err.stack);
-      return res.status(500).json({
-        error: 'GetJobListError',
-        message: 'get job list error',
-      });
+      return next(createError.unknown(err));
     } else if (jobList === undefined) {
+      // Unreachable
```

Comment:
If these lines are unreachable, why do we still preserve them?

Reply:
@Gerhut only refined error handling in this PR.